### ISSUE TITLE
Resuming training from checkpoint not possible

### ISSUE THREAD

**Issue by michaeldammann** [Noble-Lab/casanovo#570]
## Describe the Issue
When resuming training from a given checkpoint using `casanovo -train --model pathtomodel.ckpt traindata.mgf`, not the whole checkpoint (e.g. LR scheduler etc.) is loaded but only the weights. This means the learning rate is reset, resulting in weird behaviour in the subsequent training process. 
Training a model for x hours, stopping, then resuming training from the latest checkpoint is a common workflow e.g. in HPC settings, where jobs are limited to for example 24 hours.

## Steps To Reproduce
Train a model from scratch, stop training, resume training with the command above.

## Expected Behavior
Both should be possible: 
(1) Loading the complete checkpoint and (2) only the weights.
(2) is still helpful for e.g. finetuning Casanovo on a smaller dataset, where using the original LR scheduler might hurt.
Technically, it's a small fix, maybe add a flag in the config to choose between (1) and (2). For (1), add `ckpt_path=self.model_filename` to the trainer.fit call.
